docs: benchmark clarification (#299)
jina-ai/docarray: docs/advanced/document-store/benchmark.md

We are interested in the single query performance on the above tasks, which means tasks 2,3,4,5,6 are evaluated using one Document at a time, repeatedly. We report the average number.

### Some important notes

> We want to remark:

Reviewer: ```suggestion
We want to note the following:
```

> * **We measure vector DB interface**: We benchmark function calls from docarray, not just the underlying backend vector database. Therefore, results for a particular backend can be influenced (positively or negatively) by our interface. 

Reviewer: ```suggestion
* **Benchmarks are conducted end-to-end**: We benchmark function calls from DocArray, not just the underlying backend vector database. Therefore, results for a particular backend can be influenced (positively or negatively) by our interface. 
```

> * **We use similar underlying search algorithms but different implementations**: In this benchmark we focus on setting only parameters `ef`, `ef_construct` and `max_connections` from HNSW. Note that there are other parameters that storage backends can fix than can have a big impact on performance and might even be not easily accesible by users.

Reviewer: ```suggestion
* **We use similar underlying search algorithms but different implementations**: In this benchmark we focus on setting only parameters `ef`, `ef_construct` and `max_connections` from HNSW. Note that there are other parameters that storage backends can fix which can have a big impact on performance and might not be easily configurable by users.
```

> * **Benchmark for users, not research**: This benchmark showcases what a user can expect to get without tuning hyper-parameters of a vector database. We strongly recommend tuning them to achieve high quality results.

Reviewer: > tuning hyper-parameters
Is hyper-parameter the right terminology for vector databases ? maybe just parameters ?

Author: I think so
